alpha/scripts/removeExternalIdFromUser.php

```diff
@@ -0,0 +1,304 @@
+
+	$users = buildRowsForUsers($updatedUsers);
+	$reportRows = $users['rows'];
+
+	return writeReportToCsv($headers, $reportRows, $partnerId, $dryRun);
+}
+
+/**
+ * Retrieves users based on partner ID and a CSV file containing user IDs.
+ *
+ * @param int $partnerId
+ * @param string $userListCsv Path to the CSV file containing a list of user IDs.
+ * @return array List of users scoped to the provided CSV.
+ * @throws Exception
+ */
+function getUsersByCsv(int $partnerId, string $userListCsv = ''): array {
+
+	$puserIds = parsePuserIdsFromCsv($userListCsv);
+	$userListChunk = array_chunk($puserIds, 100);
+	$usersList = [];
+
+	foreach ($userListChunk as $puserIdsChunk) {
+		KalturaLog::log('Processing user IDs in chunk: [' . implode(',', $puserIdsChunk) . ']');
+		$usersChunk = getPUsersIn($partnerId, $puserIdsChunk);
+
+		if (!empty($usersChunk)) {
+			$usersList = array_merge($usersList, $usersChunk);
+		}
+	}
+
+	return $usersList;
+}
+
+/**
+ * Parses a CSV file to extract unique puserId values. The method assumes the CSV may include a header row
+ * with a column named "puserId". If no header is present, the first column of each row is used as the source of puserId values.
+ *
+ * @param string $userListCsv Path to the CSV file containing the user list.
+ *
+ * @return array An array of unique puserId values extracted from the CSV file.
+ *
+ * @throws Exception If the CSV file cannot be opened, is empty, or contains no valid puserId values.
+ */
+function parsePuserIdsFromCsv(string $userListCsv): array {
+	$handle = fopen($userListCsv, 'r');
+
+	if ($handle === false) {
+		throw new Exception("Failed to open user list CSV: $userListCsv");
+	}
+
+	$header = fgetcsv($handle);
+
+	if ($header === false) {
+		fclose($handle);
+		throw new Exception("User list CSV is empty: $userListCsv");
+	}
+
+	$trimmedHeader = array_map('trim', $header);
+	$lowerHeader = array_map('strtolower', $trimmedHeader);
+	$puserIdIndex = array_search('puserid', $lowerHeader, true);
+	$hasHeader = $puserIdIndex !== false;
+
+	if ($hasHeader === false) {
+		// treat first row as data and reset pointer
+		rewind($handle);
+	}
+
+	$puserIds = [];
+
+	while (($row = fgetcsv($handle)) !== false) {
+		if ($hasHeader) {
+			if (!array_key_exists($puserIdIndex, $row)) {
+				continue;
+			}
+			$puserId = trim((string) $row[$puserIdIndex]);
+		} else {
+			$puserId = trim((string) ($row[0] ?? ''));
+		}
+
+		if ($puserId !== '') {
+			$puserIds[] = $puserId;
+		}
+	}
+
+	fclose($handle);
+
+	$puserIds = array_values(array_unique($puserIds));
+
+	if (empty($puserIds)) {
+		throw new Exception("No puserId values found in CSV: $userListCsv");
+	}
+
+	return $puserIds;
+}
+
+function getPUsersIn($partnerId, array $puserIds = []): array {
+	if (empty($puserIds)) {
+		return [];
+	}
+
+	$puserCriteria = new Criteria();
+	$puserCriteria->add(kuserPeer::PARTNER_ID, $partnerId, Criteria::EQUAL);
+	$puserCriteria->add(kuserPeer::STATUS, KuserStatus::DELETED, Criteria::NOT_EQUAL);
```

Comment:
@orekalt no need to set STATUS criteria- kuserPeer sets this as part of its default criteria. 
https://github.com/kaltura/server/blob/67c468c731173d45a918a809f0beed0d4fd49e98/alpha/lib/model/om/BasekuserPeer.php#L572